### PR TITLE
SAK-497 Fetching all sections for publication

### PR DESCRIPTION
**Pull request by rafalkanski**
@filipgolonka @wsromek @arkadiuszmalina @bartosz-kahl 

**DONE:**
`Scenario: Fetching all sections for publication`

### CODE REVIEW

**Comment by arkadiuszmalina** [comment]
👍

**Comment by barkahl** [comment]
👍

File: docs/USAGE.md [hunk 1]
```diff
@@ -2,38 +2,51 @@
...
+
+### Fetching all sections for publication id
+
+```
+$sectionRepository = new SectionRepository($httpClient);
+
+try {
+    $findParameters = FindParameters::createForPublicationId('sa');
+    $sections = $sectionRepository->findAll($findParameters);
+} catch (CouldNotFetchSectioneRepositoryException $exception) {
```
Comment (filipgolonka) [inline docs/USAGE.md]: typo, should be `SectionRepository`

File: features/bootstrap/Snt/Capi/ArticleRepositoryContext.php [hunk 2]
```diff
@@ -102,37 +100,21 @@ public function iAskForArticlesForPublicationUsingArticleRepository($publication
...
-        } else {
-            $message = sprintf(
-                self::ARTICLE_NOT_FOUND_EXCEPTION_MESSAGE,
-                $publicationId,
-                $articleId
-            );
-
-            throw new RuntimeException($message);
-        }
+        PhpUnit::assertEquals($this->articlesFromApi[$publicationId][$articleId], $this->articles[$articleId]);
```
Comment (filipgolonka) [inline features/bootstrap/Snt/Capi/ArticleRepositoryContext.php:107]: why PhpUnit? can't we use `phpspec-expect` here?

File: src/Snt/Capi/Repository/Section/SectionRepository.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            );
+        } catch (HttpException $exception) {
+            throw new CouldNotFetchSectionRepositoryException(
+                $exception->getMessage(),
+                $exception->getCode(),
+                $exception
+            );
+        }
+
+        return $sectionsRawData['sections'];
```
Comment (filipgolonka) [inline src/Snt/Capi/Repository/Section/SectionRepository.php]: what if `sections` key does not exist in response from httpClient?

**Comment by filipgolonka** [comment]
:+1:

File: src/Snt/Capi/Repository/Section/SectionRepository.php (hunk 3 above)
Comment (rafalkanski) [inline src/Snt/Capi/Repository/Section/SectionRepository.php]: i planned that in next cycle (PR has almost 500 lines)

File: features/bootstrap/Snt/Capi/ArticleRepositoryContext.php (hunk 2 above)
Comment (rafalkanski) [inline features/bootstrap/Snt/Capi/ArticleRepositoryContext.php:107]: i like phpunit :|

**Comment by rafalkanski** [comment]
@filipgolonka review

**Comment by filipgolonka** [comment]
:+1:

**Pull request merged**